storage/innobase/dict/dict0stats.cc

```diff
@@ -1826,9 +1918,10 @@ dict_stats_scan_page(
 		placeholders). */
 		std::swap(offsets_rec, offsets_next_rec);
 
-		if (should_count_external_pages) {
-			*n_external_pages += btr_rec_get_externally_stored_len(
-				rec, offsets_rec);
+		if (leaf) {
+			leaf_stats->n_external_pages +=
+				btr_rec_get_externally_stored_len(
+					rec, offsets_rec);
```

Comment:
The function is only being invoked from this compilation unit. Better move the definition from `btr0cur.cc` to this file and make it `static`.